### PR TITLE
WIP Explain view

### PR DESCRIPTION
**Pull request by jamii**

### CODE REVIEW

File: src/catalog/lib.rs [hunk 1]
```diff
@@ -87,7 +87,7 @@ pub enum SchemaType {
 
 #[derive(Clone, Debug)]
 pub struct CatalogEntry {
-    inner: CatalogItem,
+    pub inner: CatalogItem,
```
Comment (benesch) [inline src/catalog/lib.rs]: This shouldn't be `pub`. There's an accessor method `item()` you can use. (Feel free to rename this field to `item` if you'd like; I assume that's part of the reason you didn't find the accessor method.)

File: src/sql/statement.rs [hunk 2]
```diff
@@ -1314,9 +1314,20 @@ fn handle_explain(
...
     params: &Params,
 ) -> Result<Plan, failure::Error> {
-    let (relation_expr, _desc, _finishing) = match explainee {
-        Explainee::View(name) => panic!("TODO(jamii)"),
-        Explainee::Query(query) => handle_query(scx, *query, params, QueryLifetime::OneShot)?,
+    let relation_expr = match explainee {
+        Explainee::View(name) => {
+            let full_name = scx.resolve_name(name.clone())?;
+            let entry = scx.catalog.get(&full_name)?;
+            match &entry.inner {
```
Comment (benesch) [inline src/sql/statement.rs]: `match entry.item()`

**Comment by benesch** [comment]
Welp, this made me think of a bug: https://github.com/MaterializeInc/materialize/issues/2395.

**Comment by jamii** [comment]
> having to keep track of both the unoptimized and optimized plans

What is the reason for the optimizer not being accessible from the planner? It would make things a lot simpler if we could just keep the sql string and the eval env and trust everything else to be deterministic.

**Comment by jamii** [comment]
Although optimization might eventually be pretty expensive, so maybe we do want to record all the stages :shrug:

**Comment by benesch** [comment]
> What is the reason for the optimizer not being accessible from the planner? It would make things a lot simpler if we could just keep the sql string and the eval env and trust everything else to be deterministic.

No particular reason! It just presently isn't. :D 

> Although optimization might eventually be pretty expensive, so maybe we do want to record all the stages

Yeah it might already be prohibitively expensive given the way `uses()` will recurse down a large view graph

**Comment by frankmcsherry** [comment]
> No particular reason! It just presently isn't. :D

There's the issue that the optimizer has access to currently installed arrangements. Does that cause a problem?

**Comment by jamii** [comment]
> Does that cause a problem?

Maybe. Doesn't the planner also know those?

**Comment by benesch** [comment]
Yep

**Pull request merged**